File: 2193/2193.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minimumNumberOfMovesToMakePali = function(root) {
    if (!root) return 0;
    return 1 + Math.max(minimumNumberOfMovesToMakePali(root.left), minimumNumberOfMovesToMakePali(root.right));
};

File: 2193/2193.go
func minimumNumberOfMovesToMakePali(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := minimumNumberOfMovesToMakePali(root.Left)
	r := minimumNumberOfMovesToMakePali(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}